gqlopentelemetry/handler.go

```diff
@@ -0,0 +1,123 @@
+package gqlopentelemetry
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/99designs/gqlgen/graphql/handler/extension"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+const instrumentationName = "github.com/99designs/gqlgen-contrib/gqlopentelemetry"
+
+// Handler is the extension to add OpenTelemetry Tracing
+type Handler struct {
+	ComplexityExtensionName string
+
+	Tracer trace.Tracer
+}
+
+var _ interface {
+	graphql.HandlerExtension
+	graphql.OperationInterceptor
+	graphql.FieldInterceptor
+} = Handler{}
+
+// ExtensionName is the name of the extension
+func (a Handler) ExtensionName() string {
+	return "OpenTelemetry"
+}
+
+// Validate validates the graphql schema
+func (a Handler) Validate(schema graphql.ExecutableSchema) error {
+	return nil
+}
+
+// InterceptOperation runs tracing on every graphql operation
+func (a Handler) InterceptOperation(ctx context.Context, next graphql.OperationHandler) graphql.ResponseHandler {
+	ctx, span := a.getTracer().Start(ctx, operationName(ctx))
+	defer span.End()
+
+	oc := graphql.GetOperationContext(ctx)
+	span.SetAttributes(
+		attribute.String("request.query", oc.RawQuery),
+	)
+	complexityExtension := a.ComplexityExtensionName
+	if complexityExtension == "" {
+		complexityExtension = "ComplexityLimit" // from extention package
+	}
+	complexityStats, ok := oc.Stats.GetExtension(complexityExtension).(*extension.ComplexityStats)
+	if !ok {
+		// complexity extension is not used
+		complexityStats = &extension.ComplexityStats{}
+	}
+
+	if complexityStats.ComplexityLimit > 0 {
+		span.SetAttributes(
+			attribute.Int("request.complexityLimit", complexityStats.ComplexityLimit),
+			attribute.Int("request.operationComplexity", complexityStats.Complexity),
+		)
+	}
```

Comment:
I don't think we're using this complexity extension in our implementation, so I think all of this is not needed